docs/conf.py

```diff
@@ -20,14 +25,57 @@
 # import sys
 # sys.path.insert(0, os.path.abspath('.'))
 
-from datetime import date
-
 def generateScalaNativeCurrentYear():
     ## encode/decode ugliness is to handle both python 2.7 and 3.n
     return str(date.today().year).encode("utf-8").decode("utf-8")
 
+def lastSnapshotVersion():
+    # Ensures we're inside a repo (raises if not)
+    _ensure_repo()
+    # Recosntructs version based on the last-published tag
+    tagName="last-published"
```

Comment:
do you also need to fetch the tag first, if you do shallow checkout? and if not shallow, then you don't need to hardcode previous release below, you can "git describe" it.